tests/signal_manager_test.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+. ./src/signal_manager.sh --source-only
+. ./tests/utils --source-only
+
+function suite() {
+  suite_addTest "default_handler_Test"
+  suite_addTest "non_signal_Test"
+  suite_addTest "new_signal_Test"
+  suite_addTest "signal_reset_Test"
+  suite_addTest "add_two_signals_Test"
+}
+
+function default_handler_Test() {
+  local -r original_dir="$PWD"
+  local -r expected=$'\n'"Oh no! An interruption! See ya..."
+  local output
+
+  cd "$SHUNIT_TMPDIR"
+
+  {
+    signal_manager default_interrupt_handler SIGTERM SIGINT
+    sleep infinity &
```

Comment:
Uouuu... that's something new, weirdly it does not have any entry to `infinity` in the man doc. Maybe this is a good contribution to the upstream :)

Reply:
Apparently this is not itself a feature of sleep. The thing is sleep accepts a floating point number, and `infinity` of `inf` is accepted as so. It actually truncates to a very high number. It seems like implementing an actual infinit is already [being considered](https://lists.gnu.org/archive/html/bug-gnulib/2020-02/txtd_O21xhDVT.txt)!